language/move-vm/runtime/src/native_functions.rs

```diff
@@ -173,3 +195,49 @@ impl<'a, 'b> NativeContext<'a, 'b> {
         self.interpreter.get_stack_frames(count)
     }
 }
+
+pub struct NativeContextDataStore<'a> {
+    data_store: &'a mut dyn DataStore,
+}
+
+impl<'a> DataStore for NativeContextDataStore<'a> {
```

Comment:
More specifically, why are you trying to implement DataStore here? Is there a specific reason for this?

Reply:
line 136 requires impl DataStore. Unfortunately a dyn cannot be converted to an impl, but a dyn can be placed in something that can be made an impl.

Reply:
And why do you need this `get_resource`? This function should not be made available to natives. As I tried to explain now multiple times, native functions do not have a clean way to read resources as this hacks the borrow semantics. The max which can be cleanly done is exists_at. Why can't just add a single simple one-liner for exists_at which is all what is needed for objects?

Reply:
... or just drop this and land the first version of objects with a TODO for the exists_at. I followup on this later.

Reply:
Sure, we still need a converter dyn to impl. I can delete the `get_resource` to `exists_at`, but it still needs this this DataStore dyn -> impl converter. I will add some documentation on this.

The alternative is to take a dyn in everywhere instead of an impl, but that can have negative performance considerations.

Reply:
I don't get it. We are doing native context extensions all over the place so what is so special about this one here? Perhaps you missed the concept of the extensions here? I suggest you just drop this and I implement it for you. You can easily land the object PR with an open TODO for exists_at.